apps/prairielearn/src/ee/pages/instructorAiGenerateDrafts/SampleQuestionDemo.tsx

```diff
@@ -66,7 +66,7 @@ export function SampleQuestionDemo({
     });
   };
 
-  const handleGenerateNewVariant = async () => {
+  const handleGenerateNewVariant = useCallback(async () => {
```

Comment:
Essentially moving the dep of `promptId` to recreating the function. Whenever the function is recreated, the effect will fire.

Reply:
While this modification works, it doesn't feel at all idiomatic or understandable to me.

Suggestion:

- Change the initialization of the variant state to `useState(generateSampleQuestionVariant(promptId))` (populate it immediately, don't rely on the effect)
- Drop the `useEffect` call entirely here.
- Drop the `useCallback`
- Put a `key={...}` on the usage of `<SampleQuestionDemo>` (IMO this is conceptually a similar problem + solution to https://react.dev/learn/you-might-not-need-an-effect#resetting-all-state-when-a-prop-changes)

If this is unclear, let me know and I can just open a PR with the proposed change.

Reply:
This is a little unclear -- when would you call `onMathjaxTypeset`?